components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,54 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    const body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
+    return body.ticker.price;
+  } catch (error) {
+    // we don't want the user to see any errors; simply don't show the conversion amount
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+};
+
+const StepDetailsCrypto = ({ onChange, data, currency }) => {
   const intl = useIntl();
-  const [currencyType, setCurrencyType] = useState(data.currency);
+  const [cryptoCurrencyType, setCryptoCurrencyType] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
+  const [convertedAmount, setConvertedAmount] = useState(data.convertedAmount?.amount);
+  const [cryptoExchangeRate, setCryptoExchangeRate] = useState(null);
   const [touched, setTouched] = useState(false);
   const dispatchChange = (field, value) => {
     onChange({ stepDetails: { ...data, [field]: value }, stepSummary: null });
   };
+
+  const storeCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+    const exchangeRate = await getCryptoExchangeRate(cryptoCurrency.value, collectiveCurrency);
+    setCryptoExchangeRate(exchangeRate);
+  };
+
+  useEffect(() => {
+    storeCryptoExchangeRate(cryptoCurrencyType, currency);
+  }, []);
+
+  useEffect(() => {
+    setConvertedAmount(amount * cryptoExchangeRate);
+    dispatchChange('convertedAmount', { amount: convertedAmount, currency });
```

Comment:
Did you that code to make sure you get the right value? My initial feedback from https://github.com/opencollective/opencollective-frontend/pull/6787#discussion_r699085375 suggested extracting the result in a `newConvertedAmount` variable.

The reason is, the `convertedAmount` variable won't be updated before we re-render. Try this, and notice that `original` gets displayed in the console rather than `modidifed`: https://codesandbox.io/s/friendly-drake-wkbqv?file=/src/App.js

Reply:
Ops, that's correct. One thing to note though is that currently we don't use this dispatched `convertedAmount`. There was an initial design to view the converted amount in the next sections (profile selection and QR code etc) but then we decided to remove it for now. We can see that in the screenshot here; https://github.com/opencollective/opencollective/issues/4578#issue-979171345

@znarf : Do you think we should reiterate on this displaying the conversion value in the contribution summary as per the initial design suggestion ? My thinking is we should have a different design to view that in the next sections, maybe not in contribution summary. If in agreement I'll open a new design issue for that so that we could discuss and re-iterate. Otherwise I think we should remove this `dispatchChange` on converted amount as it's not used.

Reply:
>  Do you think we should reiterate on this displaying the conversion value in the contribution summary as per the initial design suggestion ?

You can create an issue if you want, no need to tackle this now.

>  I think we should remove this dispatchChange on converted amount as it's not used.

Good idea!

Reply:
Refactored the code to remove dispatching the `convertedAmount`. I've opened a new issue for design to keep track; https://github.com/opencollective/opencollective/issues/4616 👍🏽